### PR TITLE
upgrade feathers to v4

### PR DESCRIPTION
**Pull request by mdehghani**

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -40,20 +40,20 @@
       "0xd03ea8624c8c5987235048901fb614fdca89b117"
     ]
   },
-  "useReviewerWhitelist": true,
-  "useDelegateWhitelist": true,
-  "useProjectOwnerWhitelist": true,
+  "useReviewerWhitelist": false,
```
Comment (aminlatifi) [inline config/default.json:43]: Why they are changed?

File: config/default.json [hunk 2]
```diff
@@ -40,20 +40,20 @@
...
       "jwt",
       "web3"
     ],
-    "path": "/authentication",
-    "service": "users",
-    "jwt": {
+    "jwtOptions": {
       "header": {
-        "type": "access"
+        "typ": "access"
```
Comment (aminlatifi) [inline config/default.json:56]: Is it correct?

File: src/authentication.backup.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable import/no-unresolved */
```
Comment (aminlatifi) [inline src/authentication.backup.js:1]: Let's remove this file, it's available in git history

File: config/default.json (hunk 2 above)
Comment (mdehghani) [reply config/default.json:56]: Yes, it's correct. For more information follow the link below.
https://docs.feathersjs.com/api/authentication/service.html#configuration

Comment (mdehghani) [reply config/default.json:56]: Note: typ in the header options is not a typo, it is part of the JWT JOSE header specification

File: config/default.json (hunk 1 above)
Comment (mdehghani) [reply config/default.json]: It was a mistake. Resolved now.

**Comment by aminlatifi** [comment]
@mohammadranjbarz Please review and test this pull request, it's very important upgrade

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz Please review and test this pull request, it's very important upgrade

Sure I will test and review this.

@mdehghani 
Is this https://github.com/Giveth/giveth-dapp/pull/2041 still draft?

**Comment by mohammadranjbarz** [comment]
@aminlatifi  @mdehghani 
I think it's good, the tests are passing and giveth-dapp works fine ( just be cautious when merging and deploying the config should change , jwtOptions,...)

**Comment by aminlatifi** [comment]
@mohammadranjbarz So let's merge it and test on develop

**Pull request merged**